src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -49,15 +51,78 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   currentSort: string[] = [];
   includeSubgroup = false;
   collapsed = true;
-  status: 'loading' | 'loaded' | 'empty' |'error' = 'loading';
+
+  state: 'fetching' | 'ready' |'error' = 'fetching';
 
   ongoingActivity: Activity = 'none';
 
+  private dataFetching = new Subject<{ groupId?: string, includeSubgroup: boolean, sort: string[] }>();
+
+  private dataUpdating = new Subject<{ action: Action, requests: PendingRequest[] }>();
+
   constructor(
     private getRequestsService: GetRequestsService,
     private requestActionService: RequestActionsService,
     private messageService: MessageService
-  ) {}
+  ) {
+    this.dataFetching.pipe(
+      switchMap(params =>
+        merge(
+          of(fetchingState()),
+          this.getRequestsService.getPendingRequests(params.groupId, params.includeSubgroup, params.sort).pipe(map(readyState))
+        )
+      )
+    ).subscribe(
+      state => {
+        this.state = state.tag;
+        if (isReady(state)) {
+          this.requests = state.data;
+          if (this.requests.length > 0) this.collapsed = false;
+        }
+      },
+      _err => {
+        this.state = 'error';
+      }
+    );
+
+    this.dataUpdating.pipe(
+      switchMap(params =>
+        merge(
+          of({
+            ...fetchingState(),
+            action: params.action,
+            activity: params.action === 'accept' ? 'accepting' : 'rejecting' as Activity,
+          }),
+          this.processRequests(params.action, params.requests)
+            .pipe(map(result => ({
+              ...readyState(result),
+              action: params.action,
+              activity: 'none' as Activity,
+            })))
+        )
+      )
+    ).subscribe(
+      state => {
+        this.state = state.tag;
+        this.ongoingActivity = state.activity;
+        if (isReady(state)) {
+          this.displayResponseToast(
+            this.parseResults(state.data),
+            state.action === 'accept' ? 'accept' : 'reject', // still use a matching as it is "by coincidence" that the type of verb match
+            state.action === 'accept' ? 'accepted' : 'declined'
+          );
+          this.selection = [];
+          this.dataFetching.next({ groupId: this.groupId, includeSubgroup: this.includeSubgroup, sort: this.currentSort });
+        }
+      },
+      err => {
+        this.state = 'ready';
+        this.processRequestError(err);
+        this.ongoingActivity = 'none';
```

Comment:
We can see that this var is in "conflict" with the state, it is not really independent of it (the html has `state === 'fetching' || ongoingActivity !== 'none'"` which shows this as well)

Reply:
I managed to merge these two variables (`state` and `ongoingActivity`) by having the state being `'fetching' | 'accepting' | 'rejecting' | 'ready' |'error'`. This simplified most of the processes.